### ISSUE TITLE
Arguments against open source

### ISSUE THREAD

**Issue by nabsul** [78bash/open#2]
My main issue with this presentation is that it seems to argue that open source is **always** the best option. I don’t think this is true.

**Security:** You only get the security benefits of open source after you’ve published your project and it has been reviewed by enough people. When you first make your code open source, you’re actually increasing your security risk. The risk only goes down when people take the time to review the code AND report anything they find. This can leave a very serious hole in security for a while (and it may never close if the project isn’t popular).

**Speed:** If you’re building a service like facebook, linkedin, or twitter, remaining closed source often means that you can do things more “quick and dirty”. If you always publish your source code, then people can more easily find security holes and bugs to take advantage of. You have to be extra careful about bugs and security if you publish code that is running live. So in that sense, going open source can slow you down. That’s why you often see that companies open their code after it has had time to mature and “harden” in terms of security. Two great examples of this:

- https://medium.com/netflix-techblog/titus-the-netflix-container-management-platform-is-now-open-source-f868c9fb5436
- https://siliconangle.com/blog/2018/04/19/apple-makes-foundationdb-database-open-source/

**Making money vs. helping other people make money:** You argue that open sourcing your ideas opens up opportunities to be known as an expert that people come to for consultations. But what does this mean exactly, and when is it a good idea? It’s a good idea when you don’t have the resources / experience to make money from your idea directly. So, a company (or person, or startup) with the power to go into business and make money off your idea will come to you and ask you for your help in that area.

No one will pay you $100,000 for consulting unless they’re planning to make 100X or more in return. The consulting approach tends to work when the market is large. If the market is niche and small (a special software for football referees for example), then going open source might not make sense.

**Companies open source things to make money:** You’ll notice that more and more big companies open source things these days. Here are some observations about that:

- Making things open source “looks good”, gives the company a better reputation, and attracts talented developers to the company
- Google makes more than 90% of its money from search. Android helps to ensure that it has more users using google search.
- You’ll notice that Google doesn’t open source the algorithms for their search engine.
- You’ll notice that most companies open source their “side projects”. In other words, projects that they don’t directly make money from.
- If a company’s open source projects becomes popular, they benefit from free testing and improvements from the community that builds around it.
- As Microsoft moves from providing software to providing services (OneDrive, Azure, Office365, etc.), protecting their software is not as important, so they open source it.

To be clear: I still believe that open source is one of the greatest ideas ever! But as an engineer, I believe every approach has advantages and disadvantages. It’s important to be aware of them.

A great resource to hear more about the challenges of maintaining and living on open source is this podcast: https://changelog.com/rfc. The podcast contains interviews with many open source developers who talk about the challenge of working on these kinds of projects.

**Comment by holzlchr** [comment]
First of all, thanks a lot Nabsul for your contribution. It's highly appreciated.

**Let me explain my idea**
This first presentation was meant as a marketing/propaganda tool. Therefore it intentionally lacks objectivity. 

The aim is to explain the concept and convince newcomers as quickly as possible.
If you want to penetrate a new market, you can't start with objective self-criticism. 
You must affirm the qualities of your product with simple concepts and slogans.
1.- Grab the attention
2.- Explain
3.- Convince
4.- Secure

In short, this presentation is made to **sell** the concept of open-source. 

I was active in marketing for many years. And this is my contribution to the open-source community.

**Your contribution**
Nevertheless, your remarks are very true. 
I'll try to make another presentation. A sort of "after sales user manual" we can use on a second stage. As soon as we know that our public understands the concept of open source.

Thanks again

**Comment by holzlchr** [comment]
First of all, thanks a lot Nabsul for your contribution. It's highly appreciated.

**Let me explain my idea**
This first presentation was meant as a marketing/propaganda tool. Therefore it intentionally lacks objectivity. 

The aim is to explain the concept and convince newcomers as quickly as possible.
If you want to penetrate a new market, you can't start with objective self-criticism. 
You must affirm the qualities of your product with simple concepts and slogans.
1.- Grab the attention
2.- Explain
3.- Convince
4.- Secure

In short, this presentation is made to **sell** the concept of open-source. 

I was active in marketing for many years. And this is my contribution to the open-source community.

**Your contribution**
Nevertheless, your remarks are very true. 
I'll try to make another presentation. A sort of "after sales user manual" we can use on a second stage. As soon as we know that our public understands the concept of open source.

Thanks again

**Comment by nabsul** [comment]
That makes sense. Sorry for my misunderstanding of your presentation :-)
